lib/rubocop/cop/rails/pluralization_grammar.rb

```diff
@@ -0,0 +1,95 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for correct grammar when using active support's
+      # core extensions to the numeric classes
+      #
+      # @example
+      #   # bad
+      #   3.day.ago
+      #
+      #   # good
+      #   45.minutes.from_now
+      class PluralizationGrammar < Cop
+        SINGULAR_DURATION_METHODS = { second: :seconds,
+                                      minute: :minutes,
+                                      hour: :hours,
+                                      day: :days,
+                                      week: :weeks,
+                                      fortnight: :fortnights,
+                                      month: :months,
+                                      year: :years }
+
+        PLURAL_DURATION_METHODS = SINGULAR_DURATION_METHODS.invert
+
+        MSG = 'Prefer `%s.%s` for grammar reasons.'
+
+        def on_send(node)
+          receiver, method_name, *_args = *node
+          return if receiver.nil?
+          return unless duration_method?(method_name)
+          return unless literal_number?(receiver)
+          number, = *receiver
+          if singular_receiver?(number) && plural_method?(method_name)
+            add_offense(node,
+                        :expression,
+                        format(MSG, number, singularize(method_name)))
+          elsif plural_receiver?(number) && singular_method?(method_name)
+            add_offense(node,
+                        :expression,
+                        format(MSG, number, pluralize(method_name)))
+          end
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            method_name = node.loc.selector.source
```

Comment:
I don't think that you need to retrieve the `source` here. Calling `node.loc.selector` will return the method as a symbol allowing `pluralize` and `singularize` to do their lookups without converting `method_name` back to a symbol. `plural_method?` will work as designed because of its call to `to_s`.